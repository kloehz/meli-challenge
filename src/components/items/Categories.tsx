interface ICategories {
    categories: string[]
}

export const Categories = ({categories}: ICategories) => {
    return (
        <div className="categories-container">
            {
                categories.map((category) => {
                    return (
                        <h5
                            key={category}
                            className="category"
                        >
                            {category}
                        </h5>
                    )
                })
            }
        </div>
    )
}
