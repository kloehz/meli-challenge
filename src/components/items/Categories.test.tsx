import { shallow } from "enzyme"
import { Categories } from "./Categories"

describe('Categories tests', () => {
    const mockCotegories = ["Hogar, Muebles y Jardín","Bazar y Cocina","Vajilla y Artículos de Servir","Vajilla","Vasos"];
    const wrapper = shallow(<Categories categories={mockCotegories}/>);

    test('should be match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should be render te correct quantity of categories', () => {
        expect(wrapper.find('h5').length).toBe(mockCotegories.length);
    });
    
})
