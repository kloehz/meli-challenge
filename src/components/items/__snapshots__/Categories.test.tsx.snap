// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories tests should be match snapshot 1`] = `ShallowWrapper {}`;
